### PR TITLE
Test overhaul test report

### PR DESCRIPTION
**Pull request by schunkes**
# Description

This PR contains some updates for the test report generation. It is a draft for now, since we might find more things that need to be updated.

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [ ] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

**Comment by schunkes** [comment]
I changed the \lt to < in `test_onedim_rpv` because latex would complain about an undefined control sequence. I have no idea why this happens and the \ll statement works without issue.

File: docs/rst/developer_guide/testing.rst [hunk 1]
```diff
@@ -4,6 +4,10 @@ Testing and test report
 =======================
 
 Eradiate is shipped with a series of tests written with `pytest <https://docs.pytest.org/en/latest/>`_.
+Tests are grouped by their scope and folders are named to enforce a fixed order of execution for the tests.
+
+At the highest level, there is a separation of tests for mitsuba plugins which are maintained in the Eradiate codebase and tests for Eradiate code itself.
```
Comment (leroyvn) [inline docs/rst/developer_guide/testing.rst:9]:
```suggestion
At the highest level, there is a separation of tests for Mitsuba plugins which are maintained in the Eradiate codebase and tests for Eradiate code itself.
```

File: docs/rst/developer_guide/testing.rst [hunk 2]
```diff
@@ -87,3 +79,22 @@ be added and the result will look similar to this:
...
+.. code-block:: bash
+
+    pytest tests -m "regression" --artefact_dir <a directory of your choice>
+
+The ``artefact_dir`` parameter defines the output directory in which the results and plots will be placed. If the directory does not exist, it will be created. The artefact directory defaults to ``./build/test_artefacts``, which is resolved relative to the current working directory.
+
+Test report
+-----------
+
+The Eradiate test report facilities have been moved to a separate repository, which can be found on `github <https://github.com/eradiate/eradiate-test-report>`_ .
```
Comment (leroyvn) [inline docs/rst/developer_guide/testing.rst:100]:
```suggestion
The Eradiate test report facilities have been moved to a separate repository, which can be found on `GitHub <https://github.com/eradiate/eradiate-test-report>`_ .
```

File: tests/02_eradiate/02_system/test_ckd_basic.py [hunk 3]
```diff
@@ -16,6 +16,21 @@ def test_ckd_basic(modes_all_ckd):
...
     This test is designed to check that the CKD infrastructure
     (preprocessing, spectral loop, postprocessing) works.
+
+    Rationale
+    ---------
+
+    * Sensor: Distant measure covering a plane (25 angular points,
+      10^6 sample per pixel) and targeting (0, 0, 0).
+    * Illumination: Directional illumination with a zenith angle
+      :math:`\theta = 0.0°` with black body irradiance spectrum.
```
Comment (leroyvn) [inline tests/02_eradiate/02_system/test_ckd_basic.py:26]: You'll have to escape this `\t`:
```suggestion
      :math:`\\theta = 0.0°` with black body irradiance spectrum.
```

**Review by leroyvn — COMMENTED** [review]
Just a couple of typos I spotted :)

**Comment by schunkes** [comment]
Alright, I fixed those typos. If you're satisfied, you can merge this and we get on to the other issues we have :D

**Comment by leroyvn** [comment]
Thanks, let's merge this!

**Pull request merged**